### PR TITLE
Add support for repeated parameters

### LINKED ISSUE #24: Case classes with repeated parameters do not work
This should be tested, and the best solution is to probably treat the repeated parameters of type `T` as a single parameter of type `Seq[T]`.

### PR DESCRIPTION
**Pull request by joroKr21**
* A parameter of type `A*` is encoded as a `Seq[A]`
* Added a `Param.repeated` flag to `interface`, but I'm not sure if this is useful

Resolves #24

### CODE REVIEW

File: core/shared/src/main/scala/interface.scala [hunk 1]
```diff
@@ -46,6 +46,17 @@ trait Param[Typeclass[_], Type] {
...
+  /** flag indicating a repeated (aka. vararg) parameter
+    *
+    * For example, for a case class,
+    * <pre>
+    * case class Account(id: String, emails: String*)
+    * </pre>
+    * the [[Param]] instance corresponding to the `emails` parameter would be `repeated` and have a
+    * [[PType]] equal to the type `Seq[String]`. Note that only the last parameter of a case class
+    * can be repeated. */
+  def repeated: Boolean
```
Comment (propensive) [inline core/shared/src/main/scala/interface.scala]: I'm happy for this additional parameter to be here. It's an extra parameter to the interface, but in #46 I'm going to offer a way to make that optional (which might result in a tiny runtime and compile-time speedup).

File: core/shared/src/main/scala/magnolia.scala [hunk 2]
```diff
@@ -204,9 +206,9 @@ object Magnolia {
       val assignedName: TermName = TermName(c.freshName(s"${genericTypeName}Typeclass"))
       val typeSymbol = genericType.typeSymbol
       val classType = if (typeSymbol.isClass) Some(typeSymbol.asClass) else None
-      val isCaseClass = classType.map(_.isCaseClass).getOrElse(false)
-      val isCaseObject = classType.map(_.isModuleClass).getOrElse(false)
-      val isSealedTrait = classType.map(_.isSealed).getOrElse(false)
+      val isCaseClass = classType.exists(_.isCaseClass)
+      val isCaseObject = classType.exists(_.isModuleClass)
+      val isSealedTrait = classType.exists(_.isSealed)
```
Comment (propensive) [inline core/shared/src/main/scala/magnolia.scala]: Heh, no idea why this wasn't done before! ;)

File: tests/src/main/scala/tests.scala [hunk 3]
```diff
@@ -176,5 +176,13 @@ object Tests extends TestApp {
     test("serialize a value class") {
       Show.gen[Length].show(new Length(100))
     }.assert(_ == "100")
+
+    test("show a case class with repeated param") {
```
Comment (propensive) [inline tests/src/main/scala/tests.scala]: Could you include a second test which handles a repeated parameter which is also a Magnolia-derived type, please? Not likely to be a problem, but an additional test would be useful.

Comment (joroKr21) [reply tests/src/main/scala/tests.scala]: I'm sorry I don't follow. What do you mean by Magnolia-derived type?

Comment (propensive) [reply tests/src/main/scala/tests.scala]: I mean a repeated type which needs Magnolia to derive the typeclass, i.e. not a `String` (`Seq[String]`), but a `Foo` (`Seq[Foo]`) for some case-class type, `Foo`.

Comment (joroKr21) [reply tests/src/main/scala/tests.scala]: Oh I see, will do :+1:

**Comment by joroKr21** [comment]
done

**Pull request merged**